skimage/util/_map_array.py

```diff
@@ -32,7 +32,7 @@ def map_array(input_arr, input_vals, output_vals, out=None):
     # arr.reshape(-1) may be preferable."
     input_arr = input_arr.reshape(-1)
     if out is None:
-        out = np.empty(orig_shape, dtype=output_vals.dtype)
```

Comment:
At a first glance I am wondering why you changed this? Shouldn't we be able to use the faster `np.empty` here because every value is overwritten?

Reply:
Now assigment only happens if key is present in dict.